common/src/size_estimation/mod.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::chain::{Destination, SignedTransaction, Transaction, TxOutput};
+
+/// Wallet errors
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum SizeEstimationError {
+    #[error("Unsupported input destination")]
+    UnsupportedInputDestination(Destination),
+}
+
+/// Return the encoded size of an input signature
+pub fn input_signature_size(destination: &Destination) -> Result<usize, SizeEstimationError> {
+    // Sizes calculated upfront
+    match destination {
+        Destination::Address(_) => Ok(103),
+        Destination::PublicKey(_) => Ok(69),
+        Destination::AnyoneCanSpend => Ok(2),
+        Destination::ScriptHash(_) | Destination::ClassicMultisig(_) => Err(
+            SizeEstimationError::UnsupportedInputDestination(destination.clone()),
+        ),
+    }
+}
+
+/// Return the encoded size for a SignedTransaction with specified outputs and empty inputs and
+/// signatures
+pub fn tx_size_with_outputs(outputs: &[TxOutput]) -> usize {
+    let tx = SignedTransaction::new(
+        Transaction::new(1, vec![], outputs.into()).expect("should not fail"),
+        vec![],
+    )
+    .expect("should not fail");
+    serialization::Encode::encoded_size(&tx)
+}
+
+pub fn get_tx_output_destination(txo: &TxOutput) -> Option<&Destination> {
```

Comment:
This function is basically playing with fire. Please remove it.

I understand it was in the wallet, but maybe in the wallet for that use-case it's fine. Putting it in common basically endangers people using this for general purpose stuff. This is the same problem we had before in TxOutput where we had generic functions like `destination()`, and no good came out of that. It only lead to confusion similar to that of polymorphism because things look alike, but they aren't.

Reply:
You can make this function private, btw. That makes it OK since it's used only here. For the wallet, write another one (even a copy is OK, given that these have different purposes).

Reply:
changed to private